chain/network/src/config.rs

```diff
@@ -168,6 +168,68 @@ pub struct NetworkConfig {
 }
 
 impl NetworkConfig {
+    /// Here you can override the NewtworkConfig with values for the JSON config.
+    fn override_config(overrides: crate::config_json::NetworkConfigOverrides, this: Self) -> Self {
+        Self {
+            // All fields should be configurable through the JSON config.
+            // These fields are explicitly defined here to raise a warning when new fields are added to NetworkConfig.
+            // You can either keep the value from the constructor or override it with fields from NetworkConfigOverrides.
+            // If the value from the constructor is a constant, you should probably add an override for it.
+            connect_to_reliable_peers_on_startup: overrides
+                .connect_to_reliable_peers_on_startup
+                .unwrap_or(this.connect_to_reliable_peers_on_startup),
+            max_send_peers: overrides.max_send_peers.unwrap_or(this.max_send_peers),
+            routed_message_ttl: overrides.routed_message_ttl.unwrap_or(this.routed_message_ttl),
+            max_routes_to_store: overrides.max_routes_to_store.unwrap_or(this.max_routes_to_store),
+            highest_peer_horizon: overrides
+                .highest_peer_horizon
+                .unwrap_or(this.highest_peer_horizon),
+            push_info_period: overrides
+                .push_info_period_millis
+                .map(|millis| time::Duration::milliseconds(millis))
+                .unwrap_or(this.push_info_period),
+            outbound_disabled: overrides.outbound_disabled.unwrap_or(this.outbound_disabled),
+            accounts_data_broadcast_rate_limit: if let (Some(qps), Some(burst)) = (
+                overrides.accounts_data_broadcast_rate_limit_qps,
+                overrides.accounts_data_broadcast_rate_limit_burst,
+            ) {
+                rate::Limit { qps, burst }
+            } else {
+                this.accounts_data_broadcast_rate_limit
+            },
+            routing_table_update_rate_limit: if let (Some(qps), Some(burst)) = (
+                overrides.routing_table_update_rate_limit_qps,
+                overrides.routing_table_update_rate_limit_burst,
+            ) {
+                rate::Limit { qps, burst }
+            } else {
+                this.routing_table_update_rate_limit
+            },
+
+            node_addr: this.node_addr,
```

Comment:
I also considered it. It can be done by passing self and only overriding the appropriate values.
The reason that I went for this approach is, in case you add a new field to NetworkConfig, you will get an warning in this function and you will be able to decide if you want to expose it in the JSON config or not.

Reply:
That is a reasonable approach.
However, when someone adds a new field to `NetworkConfig`, why wouldn't they simply add a field here? How can they be made aware that each new field needs to be configurable, either directly, or via this emergency mechanism?

Reply:
`why wouldn't they simply add a field here`
Authors can overlook this override function.
`How can they be made aware that each new field needs to be configurable`
They will be made aware by the rustc error saying that there is a missing field in this instantiation and they will read the comment. At this point _they will decide_ if they want to keep the value or overriding it by going to `NetworkConfigOverrides` and adding the field there.